solidity/random-beacon/contracts/DKGValidator.sol

```diff
@@ -0,0 +1,230 @@
+// SPDX-License-Identifier: MIT
+
+import "@openzeppelin/contracts/utils/cryptography/ECDSA.sol";
+import "./libraries/BytesLib.sol";
+import "./libraries/DKG.sol";
+
+pragma solidity ^0.8.6;
+
+/// @title DKG result validator
+/// @notice DKGValidator allows performing a full validation of DKG result,
+///         including checking the format of fields in the result, declared
+///         selected group members, and signatures of operators supporting the
+///         result. The operator submitting the result should perform the
+///         validation using a free contract call before submitting the result
+///         to ensure their result is valid and can not be challenged. All other
+///         network operators should perform validation of the submitted result
+///         using a free contract call and challenge the result if the
+///         validation fails.
+contract DKGValidator {
+    using BytesLib for bytes;
+    using ECDSA for bytes32;
+
+    /// @dev Size of a group in the threshold relay.
+    uint256 public constant groupSize = 64;
+
+    /// @dev The minimum number of group members needed to interact according to
+    ///      the protocol to produce a relay entry. The adversary can not learn
+    ///      anything about the key as long as it does not break into
+    ///      groupThreshold+1 of members.
+    uint256 public constant groupThreshold = 33;
+
+    /// @dev The minimum number of active and properly behaving group members
+    ///      during the DKG needed to accept the result. This number is higher
+    ///      than `groupThreshold` to keep a safety margin for members becoming
+    ///      inactive after DKG so that the group can still produce a relay
+    ///      entry.
+    uint256 public constant activeThreshold = 58; // 90% of groupSize
+
+    SortitionPool public sortitionPool;
+
+    constructor(SortitionPool _sortitionPool) {
+        sortitionPool = _sortitionPool;
+    }
+
+    /// @notice Performs a full validation of DKG result, including checking the
+    ///         format of fields in the result, declared selected group members,
+    ///         and signatures of operators supporting the result.
+    /// @param seed seed used to start the DKG and select group members
+    /// @param startBlock DKG start block
+    /// @return isValid true if the result is valid, false otherwise
+    /// @return errorMsg validation error message; empty for a valid result
+    function validate(
+        DKG.Result calldata result,
+        uint256 seed,
+        uint256 startBlock
+    ) external view returns (bool isValid, string memory errorMsg) {
```

Comment:
We don't assign anything to `isValid` and `errorMsg`, so maybe:
```
external view returns (bool , string memory) {
....
}

Reply:
I was thinking about it but at the same time, we know what these fields represent if we name them - we can even use their names in docs.